core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -30,6 +31,8 @@ class SearchService(
 ) extends ClassfileIndexer
     with ClassfileListener
     with SLF4JLogging {
+
+  private val QUERY_TIMEOUT = 30 seconds
```

Comment:
hmm, timeouts here make me nervous. They will fail when the CI machines are running slowly, and sometimes the queries can be really long on a user machine. It all happens in the background so users don't typically notice (e.g. if H2 decides to recreate its memory space on disk)